modules/analytics/src/views/full/index.tsx

```diff
@@ -218,29 +216,41 @@ const Analytics: FC<any> = ({ bp }) => {
   }
 
   const fetchQnaQuestions = async () => {
-    const qnaIds = orderMetrics(getMetric('msg_sent_qna_count'))
-      .filter(m => m.name)
-      .slice(0, 10)
-      .map(m => m.name)
+    const metrics = orderMetrics(getMetric('msg_sent_qna_count').filter(metric => metric.subMetric)).slice(0, 10)
 
-    const fetchedQuestions = await Promise.all(
-      qnaIds.filter(id => !(id in qnaQuestionsCache)).map(id => fetchQnaQuestion(id.replace('__qna__', '')))
-    )
+    const topQnaQuestions = await Promise.all(
+      metrics.map(async ({ name: id, count }) => {
+        if (id in qnaQuestionsCache.found) {
+          return { count, id, question: qnaQuestionsCache.found[id] }
+        }
 
-    for (const {
-      data: { questions },
-      id
-    } of fetchedQuestions) {
-      const question = questions[lang.getLocale()] || questions[lang.defaultLocale] || Object.values(questions)[0]
-      qnaQuestionsCache[`__qna__${id}`] = question[0]
-    }
+        if (qnaQuestionsCache.notFound.has(id)) {
+          return { count, id }
+        }
 
-    const qnaQuestions = qnaIds.reduce((acc, id) => {
-      acc[id] = qnaQuestionsCache[id]
-      return acc
-    }, {})
+        let response
+        try {
+          response = await fetchQnaQuestion(id.replace('__qna__', ''))
+        } catch (e) {
```

Comment:
What if error is not 404? Do we want to swallow the error?

Reply:
I prefer keeping the same behaviour. If this ever confuses users, we'll adapt.